include/sys/socket.h

```diff
@@ -327,7 +327,7 @@ struct sockaddr_storage
 
   /* Following fields are implementation-defined */
 
-  struct
+  struct __attribute__((packed))
```

Comment:
@nuttxs please replace with begin_packed_struct/end_packed_struct

Reply:
@xiaoxiang781216 what is the advance of begin/end_packed_struct over __attribute__((packed)) ?

Reply:
__attribute__((packed)) is gcc specific extension, other compiler(e.g. msvc) doesn't support it at all.

Reply:
@xiaoxiang781216 I'm already running the Mirrored Builds twice every day, why don't we use it? https://github.com/NuttX/nuttx/actions/runs/11630100298

@nuttxs To verify the Windows Build on your own NuttX Repo, please follow the steps in "To Run Windows Jobs" thanks!
- https://github.com/apache/nuttx/issues/14407

Reply:
> @xiaoxiang781216 I'm already running the Mirrored Builds twice every day, why don't we use it? https://github.com/NuttX/nuttx/actions/runs/11630100298
> 
> @nuttxs To verify the Windows Build on your own NuttX Repo, please follow the steps in "To Run Windows Jobs" thanks!
> 
> * [[Guide] How to Verify a PR Merge #14407](https://github.com/apache/nuttx/issues/14407)

But it's too late, it's better to catch the error before merging if we have the free quota, please add one macOS, Windows and msys2 to ci. If there is no free quota, we should remove some Ubuntu ARM build from CI to free the quota to later.

Reply:
@xiaoxiang781216 Let's discuss here thanks!
- https://github.com/apache/nuttx/issues/14598

Reply:
@xiaoxiang781216 
My apologies for the late response. I noticed that you have already submitted a pull request ([PR#14608](https://github.com/apache/nuttx/pull/14608)) addressing the compatibility of __attribute__((packed)) for MSVC.

By the way, does MSVC support the **__attribute__((aligned(n)))** attribute or keyword? I couldn't find the corresponding definition in compiler.h, and Nuttx makes extensive use of **__attribute__((aligned(n)))** for alignment.

Reply:
@lupyuen 
I will refer to the guide (https://github.com/apache/nuttx/issues/14407) process description. Thanks.

Reply:
Thanks @nuttxs! We have a newer guide here:
- https://github.com/apache/nuttx/issues/14601#issuecomment-2452875114